modyn/selector/internal/selector_strategies/coreset_strategy.py

```diff
@@ -0,0 +1,94 @@
+import random
+from typing import Iterable
+
+from modyn.metadata_database.metadata_database_connection import MetadataDatabaseConnection
+from modyn.metadata_database.models import SelectorStateMetadata
+from modyn.selector.internal.selector_strategies import AbstractSelectionStrategy
+from modyn.selector.internal.selector_strategies.downsampling_strategies import (
+    DownsamplingScheduler,
+    instantiate_scheduler,
+)
+from modyn.selector.internal.selector_strategies.presampling_strategies import AbstractPresamplingStrategy
+from modyn.selector.internal.selector_strategies.presampling_strategies.utils import instantiate_presampler
+
+
+class CoresetStrategy(AbstractSelectionStrategy):
+    def __init__(self, config: dict, modyn_config: dict, pipeline_id: int, maximum_keys_in_memory: int):
+        super().__init__(config, modyn_config, pipeline_id, maximum_keys_in_memory)
+
+        # Every coreset method has a presampling strategy to select datapoints to train on
+        self.presampling_strategy: AbstractPresamplingStrategy = instantiate_presampler(
+            config, modyn_config, pipeline_id
+        )
+        # and a downsampler scheduler to downsample the data at the trainer server. The scheduler might just be a single
+        # strategy.
+        self.downsampling_scheduler: DownsamplingScheduler = instantiate_scheduler(config, maximum_keys_in_memory)
+
+    def inform_data(self, keys: list[int], timestamps: list[int], labels: list[int]) -> None:
+        assert len(keys) == len(timestamps)
+        assert len(timestamps) == len(labels)
+
+        self._persist_samples(keys, timestamps, labels)
+
+    def trigger(self) -> tuple[int, int, int]:
+        trigger_id, total_keys_in_trigger, num_partitions = super().trigger()
+        self.downsampling_scheduler.inform_next_trigger(self._next_trigger_id)
```

Comment:
Could we also do that in `_on_trigger`? Then we would not need to overwrite trigger. `on_trigger` is called once per trigger and you could potentially call it before the for loop, couldn't you?

Reply:
Actually no, because `on_trigger` is called before incrementing `next_trigger_id` but as it is written now the scheduler needs `next_trigger_id` to be already updated

Reply:
Ok, then let's leave it like that. If you added a test that `trigger` on `CoresetStrategy` ensures that inform_next_trigger is called, then feel free to merge!

Reply:
Now there is a test